app/code/Magento/Swatches/Model/Plugin/EavAttribute.php

```diff
@@ -244,25 +253,44 @@ protected function saveSwatchParams(Attribute $attribute)
      * Save Visual Swatch data
      *
      * @param Attribute $attribute
+     *
      * @return void
      */
     protected function processVisualSwatch(Attribute $attribute)
     {
         $swatchArray = $attribute->getData('swatch/value');
+
         if (isset($swatchArray) && is_array($swatchArray)) {
-            foreach ($swatchArray as $optionId => $value) {
+            $visualSwatches = [];
+            foreach ($swatchArray as $key => $value) {
+                $visualSwatches[$key] =
+                    [
+                        self::DEFAULT_STORE_ID => $value,
+                        $this->storeManager->getStore()->getId() => '',
+                    ];
+            }
+
+            foreach ($visualSwatches as $optionId => $storeValues) {
```

Comment:
Is there any specific reason to introduce another `foreach`? It looks like logic may be merged in the first one

Reply:
@ishakhsuvarov merged with another `foreach`